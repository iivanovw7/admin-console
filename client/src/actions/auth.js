import * as types from '../constants/ActionTypes';
import Cookies from 'js-cookie';
import * as URL from '../constants/APIurl';
import axios from 'axios';

export const signInAction = ({ email, password }, history) => {

  const inOneWeek = new Date(new Date().getTime() + (1000 * 60 * 60 * 24 * 7));

  return async dispatch => {

    await axios.post(`${URL.PRIVATE_API}/auth/login`, {
      email,
      password
    }, { withCredentials: true })
               .then(response => {

                 Cookies.set('LoggedUserObject', response.data, { expires: inOneWeek });
                 Cookies.set('username', response.data.name, { expires: inOneWeek });

                 dispatch({
                   type: types.AUTHENTICATED
                 });
                 history.push('/statistics');

               })
               .catch(error => {

                 dispatch({
                   type: types.AUTHENTICATION_ERROR
                 });

               });

  };
};

export const signOutAction = history => {

  return async dispatch => {

    await axios.get(`${URL.PRIVATE_API}/auth/logout`, { withCredentials: true })
               .then(response => {
                 Cookies.remove('LoggedUserObject');
                 dispatch({
                   type: types.UNAUTHENTICATED
                 });
                 history.push('/');
               })
               .catch(error => {
                 console.log(error);
                 Cookies.remove('LoggedUserObject');
                 dispatch({
                   type: types.UNAUTHENTICATED
                 });
                 history.push('/');
               });
  };

};


