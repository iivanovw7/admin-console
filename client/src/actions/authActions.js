import * as axios from 'axios';
import Cookies from 'js-cookie';
import * as types from '../constants/actionTypes';
import * as URL from '../constants/api';

export const loginUser = ({ email, password }, history) => {

  //TODO
  // consider new way of authenticating user for shorter period of time, may be use reducer
  // instead of cookies to store data and add special route to check authenticated users after
  // page got rendered
  const inOneWeek = new Date(new Date().getTime() + (1000 * 60 * 60 * 24 * 7));

  return async dispatch => {
    await axios.post(`${URL.PRIVATE_API}/auth/login`, {
      email,
      password
    }, { withCredentials: true })
               .then(response => {
                 Cookies.set('LoggedUserObject', response.data, { expires: inOneWeek });
                 Cookies.set('username', response.data.name, { expires: inOneWeek });
                 Cookies.set('id', response.data._id, { expires: inOneWeek });
                 dispatch({
                   type: types.AUTHENTICATED,
                   payload: response
                 });
                 history.push('/statistics');

               })
               .catch(error => {
                 dispatch({
                   type: types.AUTHENTICATION_ERROR
                 });
               });
  };
};

export const logoutUser = history => {

  return async dispatch => {
    await axios.get(`${URL.PRIVATE_API}/auth/logout`, { withCredentials: true })
               .then(response => {
                 Cookies.remove('LoggedUserObject');
                 Cookies.remove('username');
                 Cookies.remove('id');
                 dispatch({
                   type: types.UNAUTHENTICATED
                 });
                 history.push('/');
               })
               .catch(error => {
                 Cookies.remove('LoggedUserObject');
                 dispatch({
                   type: types.UNAUTHENTICATED
                 });
                 history.push('/');
               });
  };
};
