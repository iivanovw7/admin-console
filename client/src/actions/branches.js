import * as types from '../constants/ActionTypes';
import Cookies from 'js-cookie';
import * as URL from '../constants/APIurl';
import axios from 'axios';

export const getBranches = (page, limit, history) => {

  return async dispatch => {
    await axios({
      method: 'get',
      url: `${URL.PRIVATE_API}/branches`,
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/x-www-form-urlencoded'
      },
      params: {
        page: page || 1,
        limit: limit || 10
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({
          type: types.FETCH_BRANCHES,
          payload: response
        });
      })
      .catch(error => {
        console.log(error);
        Cookies.remove('LoggedUserObject');
        dispatch({
          type: types.UNAUTHENTICATED
        });
        history.push('/');
      });
  };

};

export const addNewBranch = formValues => {

  const { name, email, phone, fax, address, information, status } = formValues;

  return async dispatch => {
    await axios({
      method: 'post',
      url: `${URL.PRIVATE_API}/branches`,
      data: {
        name, email, phone, fax, address, information, status
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({
          type: types.ADD_BRANCH,
          payload: response
        });
      })
      .catch(error => {
        console.log(error);
        dispatch({
          type: types.ERROR
        });
      });

  };


};

export const updateBranch = (formValues, id) => {

  const { name, email, phone, fax, address, information, status } = formValues;

  return async dispatch => {
    await axios({
      method: 'put',
      url: `${URL.PRIVATE_API}/branches/${id}`,
      data: {
        name, email, phone, fax, address, information, status
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({
          type: types.UPDATE_BRANCH,
          payload: response
        });
      })
      .catch(error => {
        console.log(error);
        dispatch({
          type: types.ERROR
        });
      });

  };
};

export const getSingleBranch = (id, history) => {

  return async dispatch => {
    await axios({
      method: 'get',
      url: `${URL.PRIVATE_API}/branches/${id}`,
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/x-www-form-urlencoded'
      },
      withCredentials: true
    })
      .then(response => {
        history.push(`/branches/${id}`);
        dispatch({
          type: types.FETCH_BRANCH,
          payload: response
        });
      })
      .catch(error => {
        console.log(error);
        Cookies.remove('LoggedUserObject');
        dispatch({
          type: types.UNAUTHENTICATED
        });
        history.push('/');
      });
  };

};