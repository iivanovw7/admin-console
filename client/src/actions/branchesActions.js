import * as axios from 'axios';
import Cookies from 'js-cookie';
import * as types from '../constants/actionTypes';
import * as URL from '../constants/api';

export const getBranches = (page, limit, history) => {

  return async dispatch => {
    await axios.get(`${URL.PRIVATE_API}/branches`, {
      params: { page, limit },
      withCredentials: true
    })
               .then(response => {
                 dispatch({
                   type: types.FETCH_BRANCHES,
                   payload: response
                 });
               })
               .catch(error => {
                 console.log(error);
                 Cookies.remove('LoggedUserObject');
                 dispatch({
                   type: types.UNAUTHENTICATED
                 });
                 history.push('/');
               });
  };
};

export const addNewBranch = formValues => {

  return async dispatch => {
    await axios.post(`${URL.PRIVATE_API}/branches`, formValues, { withCredentials: true })
               .then(response => {
                 dispatch({
                   type: types.ADD_BRANCH,
                   payload: response
                 });
               })
               .catch(error => {
                 console.log(error);
                 dispatch({
                   type: types.GENERAL_ERROR
                 });
               });
  };
};

export const updateBranch = (formValues, id) => {

  return async dispatch => {
    await axios.put(`${URL.PRIVATE_API}/branches/${id}`, formValues, { withCredentials: true })
               .then(response => {
                 dispatch({
                   type: types.UPDATE_BRANCH,
                   payload: response
                 });
               })
               .catch(error => {
                 console.log(error);
                 dispatch({
                   type: types.GENERAL_ERROR
                 });
               });
  };
};

export const getSingleBranch = (id, history) => {

  return async dispatch => {
    await axios.get(`${URL.PRIVATE_API}/branches/${id}`, { withCredentials: true })
               .then(response => {
                 history.push(`/branches/${id}`);
                 dispatch({
                   type: types.FETCH_BRANCH,
                   payload: response
                 });
               })
               .catch(error => {
                 console.log(error);
                 dispatch({
                   type: types.FETCH_ERROR
                 });
                 history.push('/branches');
               });
  };
};