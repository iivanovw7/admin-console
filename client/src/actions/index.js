export * from './authActions';
export * from './branchesActions';
export * from './groupsActions';
export * from './rolesActions';
export * from './usersActions';
export * from './ticketsActions';
export * from './messagesActions';
export * from './statisticsActions';