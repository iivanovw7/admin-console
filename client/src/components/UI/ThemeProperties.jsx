import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

