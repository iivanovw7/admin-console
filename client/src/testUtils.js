import checkPropTypes from 'check-prop-types';
import { createStore } from 'redux';
import rootReducer from './reducers';

/**
 * @param {object} initialState - initial state to store
 * @function storeFactory
 * @returns {Store} Redux store
 */
export const storeFactory = initialState => {
  return createStore(rootReducer, initialState);
};

/**
 * Returns node(s) with the given test attributes
 * @param {object} wrapper - shallow wrapper
 * @param {string} val - data-test attribute to find
 * @returns {wrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test='${val}']`);
};