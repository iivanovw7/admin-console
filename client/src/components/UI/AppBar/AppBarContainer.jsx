import { AppBar, IconButton, Toolbar, Typography } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import MenuIcon from '@material-ui/icons/Menu';
import Cookies from 'js-cookie';
import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { logoutUser } from '../../../actions';
import { NavigationStyles } from '../ThemeProperties';
import { LogoutButton } from './LogoutButton';

const AppBarContainer = props => {
  const { classes, handleDrawerToggle, history, dispatch } = props;

  return (
    <AppBar position="fixed" className={classes.appBar}>
      <Toolbar className={classes.toolbar}>
        <IconButton
          color="inherit"
          aria-label="Open drawer"
          onClick={handleDrawerToggle}
          className={classes.menuButton}
        >
          <MenuIcon/>
        </IconButton>
        <div/>
        <Typography className={classes.toolbarUserName} variant="h6" color="inherit" noWrap>
          {Cookies.get('username')}
          &nbsp;
          <LogoutButton
            handleLogout={logoutUser}
            {...props}
          />
        </Typography>
      </Toolbar>
    </AppBar>
  );
};

AppBarContainer.propTypes = {
  classes: PropTypes.object.isRequired,
  handleDrawerToggle: PropTypes.func
};

function mapStateToProps(state) {
  return { user: state.auth.user.loggedUserObject };
}

export default connect(mapStateToProps, { logoutUser })(withStyles(NavigationStyles, { withTheme: true })(withRouter(AppBarContainer)));
