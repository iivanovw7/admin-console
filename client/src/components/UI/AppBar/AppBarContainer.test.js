import { Icon, withStyles } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import jest from 'jest-mock';
import React from 'react';
import { connect, Provider } from 'react-redux';
import { MemoryRouter, withRouter } from 'react-router-dom';
import { logoutUser } from '../../../actions';
import AppBarContainer from '../AppBar/AppBarContainer';
import { NavigationStyles } from '../ThemeProperties';
import { history, storeFactory } from '../../../testUtils';

const setup = (initialState = {}) => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  let store = storeFactory(initialState);

  const Composition = connect(
    null, { logoutUser }
  )(withStyles(NavigationStyles, { withTheme: true })(withRouter(AppBarContainer)));

  const props = {
    history: history
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  /**
   *  Renders AppBar container with styles
   */
  describe('AppBar container', () => {
    it('Should render AppBar container with fields and styles', () => {
      const mockFunc = jest.fn();
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <Composition {...props} dispatch={mockFunc}/>
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.find(Icon).text()).toBe('exit_to_app');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
};

setup();
