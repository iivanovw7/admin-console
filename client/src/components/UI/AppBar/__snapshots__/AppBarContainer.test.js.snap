// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar container Should render AppBar container with fields and styles 1`] = `"<header class=\\"MuiPaper-root-38 MuiPaper-elevation4-44 MuiAppBar-root-29 MuiAppBar-positionFixed-30 MuiAppBar-colorPrimary-36 mui-fixed withRouter-AppBarContainer--appBar-18 withRouter-Connect-WithStyles-withRouter-AppBarContainer-----appBar-4\\"><div class=\\"MuiToolbar-root-65 MuiToolbar-regular-67 MuiToolbar-gutters-66 withRouter-AppBarContainer--toolbar-20 withRouter-Connect-WithStyles-withRouter-AppBarContainer-----toolbar-6\\"><button class=\\"MuiButtonBase-root-75 MuiIconButton-root-69 MuiIconButton-colorInherit-70 withRouter-AppBarContainer--menuButton-19 withRouter-Connect-WithStyles-withRouter-AppBarContainer-----menuButton-5\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Open drawer\\"><span class=\\"MuiIconButton-label-74\\"><svg class=\\"MuiSvgIcon-root-78\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span><span class=\\"MuiTouchRipple-root-132\\"></span></button><div></div><h6 class=\\"MuiTypography-root-87 MuiTypography-h6-104 MuiTypography-colorInherit-116 MuiTypography-noWrap-113 withRouter-AppBarContainer--toolbarUserName-21 withRouter-Connect-WithStyles-withRouter-AppBarContainer-----toolbarUserName-7\\">&nbsp;<button class=\\"MuiButtonBase-root-75 MuiIconButton-root-69\\" tabindex=\\"0\\" type=\\"button\\" style=\\"text-transform: none; color: white;\\" aria-label=\\"Delete\\"><span class=\\"MuiIconButton-label-74\\"><span class=\\"material-icons MuiIcon-root-123\\" aria-hidden=\\"true\\">exit_to_app</span></span><span class=\\"MuiTouchRipple-root-132\\"></span></button></h6></div></header>"`;
